be/src/exprs/expr_context.cpp

```diff
@@ -42,8 +42,9 @@ ExprContext::ExprContext(Expr* root)
         : _fn_contexts_ptr(nullptr), _root(root), _is_clone(false), _prepared(false), _opened(false), _closed(false) {}
 
 ExprContext::~ExprContext() {
-    DCHECK(!_prepared || _closed) << ". expr context address = " << this;
+    // DCHECK(!_prepared || _closed) << ". expr context address = " << this;
```

Comment:
why we remove this check

Reply:
This check is no reasonable. It implies the `ExprContext` must be closed before deconstruct. But some local variable, this requirement is very hard to meet.

The example is `be/src/storage/column_expr_predicate.cpp`.

Reply:
I think it is necessary to call close manually

Reply:
if we create a state in state in prepare state we should close it in FRAGMENT_LOCAL STAGE

Reply:
Why call the `close` manually? If we could make sure the memory would not leak.

Could you give me some more reason?

Reply:
when we will alloc object in regex_prepare and free it in regex_close